### DISCUSSION TITLE
Validation: "Must not match" Element of Array in Number Field

### DISCUSSION THREAD

**Discussion by Doctor-N0** [Budibase/budibase#13738] [Help] [answered]
Is it possible to create a custom validation rule for a number field that takes an array as binding input, such as `[1,2,3]`, and if the entered number by the user (e.g., `2`) is one of the elements in the array, display an error message (e.g., "Please enter a value that is not within [1,2,3]")?

**Comment by mikesealey** [comment, marked as answer]
Hi @Doctor-N0 

I can see you've asked a similar question before. I created a feature-request on your behalf, but forgot to get back to your original discussion. Here's a [link to the feature request](https://github.com/Budibase/budibase/issues/13541), you're welcome to drop in any additional notes that you think should be considered when it comes to developing that feature.

As for a solution that's currently available in Budibase, there are a few ways to approach it, and they all boil down to building your own validation in state. I worked on a similar issue that you can see [here](https://github.com/Budibase/budibase/discussions/13383), where the solution bears some similarities.

In short you can use state to store the values of the array and either use some JavaScript in the bindings menu, or use an action to trigger an automation that runs some JavaScript. 

The JavaScript you need to run is really straightforward. The `.includes()` method is used on an array, and given an element as an argument. It will return a boolean (true or false). Read more about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

With that boolean you can then apply some conditions to the visibility of a container that might show or hide a warning or error, as in the example.
![image](https://github.com/Budibase/budibase/assets/39269380/209e3987-9be4-454e-a71f-94452da5dcb1)


If you want to use that boolean to also prevent the submission of the form you can use a `Continue If/Stop If` action
![image](https://github.com/Budibase/budibase/assets/39269380/2f1d0f2b-2618-4acc-9611-1392a4152805)

I hope this helps get you unstuck. Keep an eye on the feature request and future releases to see if more custom validation options become available.

**Reply by Doctor-N0** [reply]
Thanks, that's a smart solution.
